lib/galaxy/tools/__init__.py

```diff
@@ -2185,6 +2184,13 @@ def handle_input(
             collection_info=collection_info,
             completed_jobs=completed_jobs,
         )
+
+        tags = incoming.get("tags", [])
+        if isinstance(tags, list) and tags:
+            tag_handler = self.app.tag_handler
```

Comment:
You'll want to use the tag handler session via trans.tag_handler here:
```suggestion
            tag_handler = trans.tag_handler
```
or if that's not the right kind of trans, establish a new one via (self.app.tag_handler.create_tag_handler_session)